functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/JobActivityHistorySpec.groovy

```diff
@@ -0,0 +1,94 @@
+
+package org.rundeck.tests.functional.selenium.jobs
+
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.WebDriverWait
+import org.rundeck.util.annotations.SeleniumCoreTest
+import org.rundeck.util.container.SeleniumBase
+import org.rundeck.util.gui.pages.login.LoginPage
+import org.rundeck.util.gui.pages.execution.ExecutionShowPage
+import org.rundeck.util.gui.pages.jobs.JobListPage
+import org.rundeck.util.gui.pages.activity.ActivityPage
+import org.rundeck.util.common.jobs.JobUtils
+import java.time.Duration
+
+@SeleniumCoreTest
+class JobActivityHistorySpec extends SeleniumBase {
+    WebDriverWait wait
+    String jobId
+
+    /**
+     * Setup the project for the test.
+     */
+    def setupSpec() {
+        setupProject(SELENIUM_BASIC_PROJECT)
+    }
+
+    /**
+     * Setup the test environment.
+     */
+    def setup() {
+        // Set up WebDriverWait to wait up to 30 seconds for elements that take time to appear or change
+        wait = new WebDriverWait(getDriver(), Duration.ofSeconds(30))
+        def loginPage = go LoginPage
+        loginPage.login(TEST_USER, TEST_PASS)
+        def jobDefinition = JobUtils.generateScheduledExecutionXml("TestJobActivityHistory")
+        def client = getClient()
+        def response = JobUtils.createJob(SELENIUM_BASIC_PROJECT, jobDefinition, client)
+        assert response.succeeded.size() > 0: "Failed to create job"
+        jobId = response.succeeded[0].id
+        def runResponse = JobUtils.executeJob(jobId, client)
+        assert runResponse.code() == 200: "Failed to execute job"
+
+    }
+
+
+    def "review job activity history from Job List Page"() {
+        when: "Navigate to the Job List Page"
+        def activityPage = go ActivityPage, SELENIUM_BASIC_PROJECT
+        activityPage.loadActivityPageForProject(SELENIUM_BASIC_PROJECT)
+        then: "Validate job execution is listed in Activity History"
+        activityPage.validatePage()
+        List<WebElement> activityList = activityPage.getActivityRows()
+        assert activityList.size() > 0: "No activity rows found"
```

Comment:
```suggestion
        !activityList.empty()
```
In general, avoid using `size()` to check for emptiness for perf reasons

Also, in a Spock specification, you do not necessarily need to use `assert` statements in the `then` section. Spock's then block is designed for assertions, and it implicitly treats any boolean expression as an assertion.